components/brave_wallet/browser/zcash/zcash_create_orchard_to_orchard_transaction_task.cc

```diff
@@ -70,28 +71,42 @@ void ZCashCreateOrchardToOrchardTransactionTask::WorkOnTask() {
 }
 
 void ZCashCreateOrchardToOrchardTransactionTask::GetSpendableNotes() {
+  if (!context_.account_internal_addr) {
+    error_ = "No internal address provided";
+    ScheduleWorkOnTask();
+    return;
+  }
   context_.sync_state->AsyncCall(&OrchardSyncState::GetSpendableNotes)
-      .WithArgs(context_.account_id.Clone())
+      .WithArgs(context_.account_id.Clone(),
+                context_.account_internal_addr.value())
       .Then(base::BindOnce(
           &ZCashCreateOrchardToOrchardTransactionTask::OnGetSpendableNotes,
           weak_ptr_factory_.GetWeakPtr()));
 }
 
 void ZCashCreateOrchardToOrchardTransactionTask::OnGetSpendableNotes(
-    base::expected<std::vector<OrchardNote>, OrchardStorage::Error> result) {
+    base::expected<std::optional<OrchardSyncState::SpendableNotesBundle>,
+                   OrchardStorage::Error> result) {
   if (!result.has_value()) {
     error_ = result.error().message;
     ScheduleWorkOnTask();
     return;
   }
 
-  spendable_notes_ = result.value();
+  if (!result.value()) {
+    error_ = "No spendable notes";
+    ScheduleWorkOnTask();
+    return;
+  }
+
+  spendable_notes_ = std::move(**result);
```

Comment:
Let's avoid double `**`. Something like  `*result.value()` would be preffered.

Reply:
`base::expected<std::optional<OrchardSyncState::SpendableNotesBundle>, OrchardStorage::Error>`
Is is possible to avoid `optional` in this type and add some sort of flag into `SpendableNotesBundle`?
Having expected over optional is very counter-intutivie

Reply:
This pattern is already widely used in the OrchardSyncState and OrchardStorage. I'd prefer keep it consistent here.